pkg/auth/providers/azure/migrate.go

```diff
@@ -0,0 +1,87 @@
+package azure
```

Comment:
Speaking generally on this new file: I am personally not a fan of when methods defined for structs are split between several files. It makes it hard to track which methods are for which structs.

But that's just me.

Reply:
I think it's fine, because the migration code is really separate, so it deserves its own file. And it's an important distinct operation/feature of the auth provider, so it's OK to designate it with a file. And the provider already has a lot of methods anyways. The actions file has some methods for it...

Reply:
Not sure that I agree that further splitting the code is a good idea.

That being said, this is largely a style item, so I'm going to resolve the discussion.

Reply:
Nit: 
I am also not a fan of splitting up methods and structs. I feel like it hurts readability ,because without an Iquicklyre is no way to easily find the struct definition or other methods the struct is using. In general when I feel the need to create separate files it usually means my struct is getting heavy and can probably be broken up. You don't have to change it for this review, but I think it's worth noting.

Reply:
I unresolved just so you can see my comment, but like I said you can leave it how you have it now.

Reply:
I'd leave it, although I completely agree with you and Michael here. I just think that here this does not hurt, because of how distinct the migrate action is. As I worked, I often searched for functions and methods of this migration process and figured that it does warrant its own file in the end.